Sources/StreamVideo/Utils/Timers/Protocol/Timer.swift

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright © 2025 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+/// A protocol describing a timing utility for scheduling one-shot and repeating
+/// timers, as well as retrieving the current time or publishing timer events.
+///
+/// All scheduled timers are executed on the provided dispatch queue, which
+/// allows running timers off the main thread in background contexts.
+protocol Timer {
```

Comment:
Is this the same implementation we have on chat? Since both products should use the same implementation eventually, from StreamCore.